sdk/core/azure-core-test/README.md

```diff
@@ -1,108 +1,289 @@
-
-* Create live or playback JUnit tests using [TestBase][TestBase.java] and
-  leveraging [InterceptorManager][InterceptorManager.java] to keep track of
-  network calls.
-* Record network calls using pipeline policy,
-  [RecordNetworkCallPolicy][RecordNetworkCallPolicy.java].
-* Playback test session records with [PlaybackClient][PlaybackClient.java].
+This library contains core classes used to test Azure SDK client libraries.
+
+Newer SDK tests utilize the [Azure SDK Tools Test Proxy][test-proxy-readme] to record and playback HTTP interactions.
+To migrate from existing [TestBase][TestBase.java] to use the test proxy, or to learn more about using the test proxy,
+refer to the [test proxy migration guide][test-proxy-migration-guide].
+
+## Table of contents
+
+- [Getting started](#getting-started)
+- [Key concepts](#key-concepts)
+- [Write or run tests](#write-or-run-tests)
+    - [Set up test resources](#set-up-test-resources)
+    - [Configure credentials](#configure-credentials)
+    - [Start the test proxy server](#start-the-test-proxy-server)
+    - [Write your tests](#write-your-tests)
+    - [Configure live or playback testing mode](#configure-live-or-playback-testing-mode)
+    - [Run and record tests](#run-and-record-tests)
+    - [Run tests with out-of-repo recordings](#run-tests-with-out-of-repo-recordings)
+    - [Sanitize secrets](#sanitize-secrets)
+- [Troubleshooting](#troubleshooting)
+- [Next steps](#next-steps)
+- [Contributing](#contributing)
 
 ## Getting started
 
 To use this package, add the following to your _pom.xml_.
 
 [//]: # ({x-version-update-start;com.azure:azure-core-test;current})
+
 ```xml
+
 <dependency>
-  <groupId>com.azure</groupId>
-  <artifactId>azure-core-test</artifactId>
-  <version>1.18.0</version>
+    <groupId>com.azure</groupId>
+    <artifactId>azure-core-test</artifactId>
+    <version>1.18.0</version>
 </dependency>
 ```
+
 [//]: # ({x-version-update-end})
 
 ## Key concepts
 
-* [TestBase][TestBase.java]: Base test class that creates an `InterceptorManager` for each unit test and either plays
-  back test session data or records the test session. Each session's file name is determined using the name of the test.
-* [InterceptorManager][InterceptorManager.java]: Main class used to record and playback `NetworkCallRecords`.
+* Run tests in `Record` mode:
+  To record means to intercept any HTTP request, store it in a file, then store the response received from the live
+  resource that was originally targeted.
+* Run tests in `Playback` mode:
+  To playback means to intercept any HTTP request and to respond it with the stored response of a previously recorded
+  matching request.
+* Run tests in `Live` mode:
+  To run live means to not intercept any HTTP request and to send them to the live resource.
+* Sanitize sensitive information:
+  Sensitive information means content like passwords, unique identifiers or personal information should be cleaned up
```

Comment:
Do we have documentation on how to sanitize sensitive data? We should add a link to the documentation or elaborate here on how do it.

Reply:
Added in detailed section.